internal/compiler/generator/rust.rs

```diff
@@ -1785,7 +1785,7 @@ fn compile_expression(expr: &Expression, ctx: &EvaluationContext) -> TokenStream
                     ))
                 }
                 (Type::Float32, Type::Model) | (Type::Int32, Type::Model) => {
-                    quote!(slint::private_unstable_api::re_exports::ModelRc::new(#f as usize))
+                    quote!(slint::private_unstable_api::re_exports::ModelRc::new(#f.max(Default::default()) as usize))
```

Comment:
In rust, the value is ensured in the generated code,  and in C++ this is the model implementation's row_count.
Would be nice if it was consistent...  (`impl Model for usize`'s `row_count` could be changed.  Or the generated C++ code could be changed,  or the constructor for the C++ model

Reply:
I think the way this is done on the Rust side is correct, the "error" is caught as early as possible. Otherwise we'd have to impl `Model` for `i32` and `f64` and **then** handle that in `row_count()`.

I'll try to fix this in C++ by changing `IntModel` (to be a `UIntModel` or so) to store an unsigned value and catch the negativity at construction time like in Rust.